packages/relay-kit/src/packs/safe-4337/Safe4337Pack.ts

```diff
@@ -191,41 +214,90 @@ export class Safe4337Pack extends RelayKitBasePack<{
       if (isApproveTransactionRequired) {
         const { paymasterAddress, amountToApprove = MAX_ERC20_AMOUNT_TO_APPROVE } = paymasterOptions
 
-        const enable4337ModulesTransaction = {
-          to: addModulesLibAddress,
-          value: '0',
-          data: INTERFACES.encodeFunctionData('enableModules', [[safe4337ModuleAddress]]),
-          operation: OperationType.DelegateCall // DelegateCall required for enabling the 4337 module
-        }
-
+        // second transaction: approve ERC-20 paymaster token
         const approveToPaymasterTransaction = {
           to: paymasterTokenAddress,
           data: INTERFACES.encodeFunctionData('approve', [paymasterAddress, amountToApprove]),
           value: '0',
           operation: OperationType.Call // Call for approve
         }
 
-        const setupBatch = [enable4337ModulesTransaction, approveToPaymasterTransaction]
+        setupTransactions.push(approveToPaymasterTransaction)
+      }
 
-        const batchData = INTERFACES.encodeFunctionData('multiSend', [
-          encodeMultiSendData(setupBatch)
-        ])
+      // third transaction: passkey support via shared signer SafeWebAuthnSharedSigner
+      // see: https://github.com/safe-global/safe-modules/blob/main/modules/passkey/contracts/4337/experimental/README.md
+      const isPasskeySigner = signer && typeof signer !== 'string'
+
+      if (isPasskeySigner) {
+        const safeProvider = new SafeProvider({
+          provider
+        })
+
+        const safeWebAuthnSignerFactoryContract = await getSafeWebAuthnSignerFactoryContract({
```

Comment:
Shouldnt we validate if this doesnt return? I understand that this function and not by the deployments and that, if it reaches this point, it means that we are probably in Sepolia (since it requires the deployment of 4337) but will that always be true?

Reply:
mmm good catch!

The `getSafeWebAuthnSignerFactoryContract` function throws an error if the contract is not deployed in the current chain.

```typescript
  if (!isContractDeployed) {
    throw new Error('safeWebAuthnSignerFactory contract is not deployed on the current network')
  }
```

However, after some research, I've realized that the current implementation only works when the `WebAuthnSignerFactory` contract is deployed.  (now its only deployed in `Sepolia`, and the address hardcoded in the code...).

To address this, I've removed the `getSafeWebAuthnSignerFactoryContract` call from the `ContractManager` (used in the `protocol-kit` instantiation). We now only invoke `getSafeWebAuthnSignerFactoryContract` when we need that specific contract (only in the passkey flows) rather than always calling it as before. 

See this commit for further details:

```
https://github.com/safe-global/safe-core-sdk/pull/841/commits/dd0dddc91d7382528a1acdf6369adbb9cac1ab69
```